client/client.go

```diff
@@ -0,0 +1,144 @@
+package client
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/cenkalti/backoff/v4"
+	"go.uber.org/zap"
+)
+
+var logger *zap.Logger
+
+func init() {
+	var err error
+	logger, err = zap.NewProduction()
+	if err != nil {
+		panic(fmt.Sprintf("Failed to initialize logger: %v", err))
+	}
+}
+
+// SetupEnv sets up the environment variables for the datetime client.
+// It takes the server URL, standard server port, and Gin server port as parameters.
+//
+// Parameters:
+//   - serverURL: The base URL of the datetime server (e.g., "http://localhost")
+//   - serverPort: The port for the standard server
+//   - serverPortGin: The port for the Gin server
+func SetupEnv(serverURL, serverPort, serverPortGin string) {
+	os.Setenv("SERVER_URL", serverURL)
+	os.Setenv("SERVER_PORT", serverPort)
+	os.Setenv("SERVER_PORT_GIN", serverPortGin)
+	logger.Info("Environment variables set",
+		zap.String("SERVER_URL", serverURL),
+		zap.String("SERVER_PORT", serverPort),
+		zap.String("SERVER_PORT_GIN", serverPortGin))
+}
+
+// GetDateTime retrieves the current date and time from the specified server.
+// It supports both standard and Gin server types and can handle JSON or plain text responses.
+//
+// Parameters:
+//   - serverType: The type of server to query ("standard" or "gin")
+//   - contentType: The desired content type for the response ("application/json" or "text/plain")
+//
+// Returns:
+//   - string: The retrieved datetime as a string
+//   - error: An error if the request fails or the response is invalid
+//
+// The function uses an exponential backoff retry mechanism for failed requests.
+func GetDateTime(serverType, contentType string) (string, error) {
+	serverURL := os.Getenv("SERVER_URL")
```

Comment:
there can be a client holding the needed values needed to get the data instead of fetching them over and over again